go/interpreter/lfvm/memory_test.go

```diff
@@ -205,38 +205,40 @@ func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
 func TestMemory_getSlice_ErrorCases(t *testing.T) {
 	c := context{gas: 0}
 	m := NewMemory()
-	_, err := m.getSlice(0, 1, &c)
+	_, err := m.getSlice(uint256.NewInt(0), uint256.NewInt(1), &c)
 	if !errors.Is(err, errOutOfGas) {
 		t.Errorf("error should be errOutOfGas, instead is: %v", err)
 	}
-	_, err = m.getSlice(math.MaxUint64-31, 32, &c)
+
+	_, err = m.getSlice(uint256.NewInt(math.MaxUint64-31), uint256.NewInt(32), &c)
 	if !errors.Is(err, errOverflow) {
 		t.Errorf("error should be errOverflow, instead is: %v", err)
 	}
 }
 
-func TestMemory_getSlice_properlyHandlesMemoryExpansionsAndReturnsExpectedMemory(t *testing.T) {
+func TestMemory_getSlice_ReturnsSliceOfSize(t *testing.T) {
 
 	tests := map[string]struct {
-		offset   uint64
-		size     uint64
+		offset   *uint256.Int
+		size     *uint256.Int
 		expected []byte
 	}{
 		"size zero returns empty slice": {
-			offset:   64,
-			size:     0,
+			offset:   uint256.NewInt(64),
```

Comment:
yes, we may want to go over the semantics of all of these uses.
I don't like neither the uses we do (by value)  vs the uses from geth (by pointer). I would prefer to have only one semantic.

Reply:
I am not sure that is possible since we cannot change how geth does some things, and sometimes we can find more optimal ways of dealing with certain things.

Reply:
We can stop using them by value. After all our stack API uses them by pointer as well.